lib/spack/spack/package.py

```diff
@@ -1552,10 +1560,15 @@ def build_process():
             # preserve verbosity across runs
             return echo
 
+        # do any checks needed by the test suite
+        # return without building if only testing
+        if not self.unit_test_check():
+            return
```

Comment:
I am not sure I get the logic of this part. Is this used only *within* unit tests to disable building?